crates/pathfinder/src/rpc/v01.rs

```diff
@@ -371,11 +372,42 @@ mod tests {
         storage::{StarknetBlock, StarknetBlocksTable, StarknetTransactionsTable, Storage},
     };
     use assert_matches::assert_matches;
+
+    use jsonrpsee::http_server::{HttpServerBuilder, HttpServerHandle};
+    use jsonrpsee::RpcModule;
     use jsonrpsee::{core::RpcResult, rpc_params, types::ParamsSer};
     use serde_json::json;
     use stark_hash::StarkHash;
+    use std::net::SocketAddr;
     use std::sync::Arc;
 
+    /// Starts a server hosting RPC v0.1 at the `/` root path with metrics middleware enabled.
+    ///
+    /// This is useful for testing, as the jsonrpsee tester assumes a root path.
+    async fn run_v01_test_server(
```

Comment:
Why not just add the ability to specify the URL to `TestClientBuilder`? `TestClient` already supports that since `target` is a `reqwest::Url` there.

Reply:
Most v0.1 tests don't actually use the `TestClientBuilder` etc. They were all just using this:
```rust
let (__handle, addr) = run_server(*LOCALHOST, api).await.unwrap();
```
so it was far easier for me to just move, rename & rewrite `run_server` as `run_v01_test_server` than to inject the `TestXXX` framework.

Reply:
Hmm, most test I've seen actually call `rpc::test_client::client(addr)` to construct a `TestClient` with the address returned by `run_server()`. I thought we could just as easily add `v01_client()` that sets up the URL properly.

Reply:
Oh hang on I misread what you meant.